### DISCUSSION TITLE
Object selection remaining in old position after position update

### DISCUSSION THREAD

**Discussion by Drealler** [fabricjs/fabric.js#9448] [Q&A]
Hey, I have this silly example of a button that sets the left position of a Rect. The position visually changes but the object is remaining in the old position.

Initial position:
![image](https://github.com/fabricjs/fabric.js/assets/14021153/3a967ef2-7f1c-4f50-9121-395a1e3669b7)

After SetLeft:
![image](https://github.com/fabricjs/fabric.js/assets/14021153/507df06d-75b4-4123-a74b-3070153b38dd)

Here I could not take a print including the mouse selection cursor, but the selection remains in the old position and the Rect is not clickable of draggable anymore. I can only unlock it by clicking in "insivible" version but then the Rect goes back to the old position after the click:
![image](https://github.com/fabricjs/fabric.js/assets/14021153/ffaa07be-cac1-4188-a3c5-54f02c579fd4)

The code (I'm using Next.js):
````
const setLeft = () => {
    const canvas = canvasInstance.current;
    const rect = rectEl.current;

    rect?.set('left', 200);
    rect?.setCoords();

    canvas?.renderAll();
  };
````

What is the correct way to update the position of objects?

**Comment by ShaMan123** [comment]
seems correct so you probably are doing something unexpected in the rest of your logic
REPRODUCTION
